app/lib/features/chat_ng/layout/chat_ng_layout_builder.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:acter/common/extensions/acter_build_context.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/features/chat_ng/pages/chat_select_page.dart';
+import 'package:acter/features/chat_ng/rooms_list/widgets/rooms_list.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+class ChatNgLayoutBuilder extends StatelessWidget {
+  final Widget? centerChild;
+  final Widget? expandedChild;
+
+  const ChatNgLayoutBuilder({this.centerChild, this.expandedChild, super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final expanded = expandedChild;
+    final center = centerChild;
+    if (!context.isLargeScreen) {
+      // we only have space to show the deepest child:
+      if (expanded != null) return expanded;
+      if (center != null) return center;
+      // no children, show the room list
+      return RoomsListNGWidget(
+        onSelected:
+            (String roomId) => context.pushNamed(
+              Routes.chatroom.name,
+              pathParameters: {'roomId': roomId},
+            ),
+      );
+    }
+
+    final pushReplacementRouting = centerChild != null;
```

Comment:
yep, use of both but same identity variables just makes this more confusing, would suggest we don't reinitialize widget parameters and use them directly like this...

Reply:
just a copy-paste, didn't change anything here...